src/targets/awsLambda.ts

```diff
@@ -0,0 +1,187 @@
+import * as fs from 'fs';
+import { logger as loggerRaw } from '../logger';
+import { TargetConfig } from '../schemas/project_config';
+import { BaseTarget } from './base';
+import { BaseArtifactProvider } from '../artifact_providers/base';
+import { ConfigurationError, reportError } from '../utils/errors';
+import { AWSError } from 'aws-sdk';
+import * as Lambda from 'aws-sdk/clients/lambda';
+import { PromiseResult } from 'aws-sdk/lib/request';
+
+const logger = loggerRaw.withScope(`[aws-lambda]`);
+
+/**
+ * RegExp for the AWS Lambda package.
+ * The pattern matches the following structure:
+ * `sentry-node-serverless-{version}.zip`.
+ */
+const DEFAULT_AWS_LAMBDA_DIST_REGEX = /sentry-node-serverless-\d+(\.\d+)*\.zip$/;
+
+const awsAllRegions = [
+  'us-east-1',
+  'us-east-2',
+  'us-west-1',
+  'us-west-2',
+  'ap-south-1',
+  'ap-southeast-1',
+  'ap-southeast-2',
+  'ap-northeast-1',
+  'ap-northeast-2',
+  'ca-central-1',
+  'eu-central-1',
+  'eu-west-1',
+  'eu-west-2',
+  'eu-west-3',
+  'sa-east-1',
+];
+
+const compatibleRuntimes = ['nodejs10.x', 'nodejs12.x'];
+
+/** Config options for the "aws-lambda" target. */
+interface AwsLambdaTargetOptions extends TargetConfig {
+  /** AWS access key ID, set as AWS_ACCESS_KEY_ID. */
+  awsAccessKeyId: string;
+  /** AWS secret access key, set as `AWS_SECRET_ACCESS_KEY`. */
+  awsSecretAccessKey: string;
+}
+
+/**
+ * The default layer name is used when no `AWS_LAYER_NAME`
+ * environment variable is found.
+ */
+export const defaultLayerName = 'SentryNodeServerlessSDK';
```

Comment:
This should be a required project-specific config, and not read from the environment.
Different projects, different layer names, no default (fail if not specified).